Index: libavformat/Makefile
===================================================================
--- libavformat/Makefile	(revision 13820)
+++ libavformat/Makefile	(working copy)
@@ -77,6 +77,7 @@
 OBJS-$(CONFIG_MJPEG_DEMUXER)             += raw.o
 OBJS-$(CONFIG_MJPEG_MUXER)               += raw.o
 OBJS-$(CONFIG_MLP_DEMUXER)               += raw.o
+OBJS-$(CONFIG_MLP_MUXER)                 += raw.o
 OBJS-$(CONFIG_MM_DEMUXER)                += mm.o
 OBJS-$(CONFIG_MMF_DEMUXER)               += mmf.o raw.o
 OBJS-$(CONFIG_MMF_MUXER)                 += mmf.o riff.o
Index: libavformat/raw.c
===================================================================
--- libavformat/raw.c	(revision 13820)
+++ libavformat/raw.c	(working copy)
@@ -455,6 +455,21 @@
     .value = CODEC_ID_MLP,
 };
 
+#ifdef CONFIG_MUXERS
+AVOutputFormat mlp_muxer = {
+    "mlp",
+    NULL_IF_CONFIG_SMALL("raw MLP"),
+    NULL,
+    "mlp",
+    0,
+    CODEC_ID_MLP,
+    CODEC_ID_NONE,
+    NULL,
+    raw_write_packet,
+    .flags= AVFMT_NOTIMESTAMPS,
+};
+#endif //CONFIG_MUXERS
+
 AVInputFormat flac_demuxer = {
     "flac",
     NULL_IF_CONFIG_SMALL("raw FLAC"),
Index: libavformat/allformats.c
===================================================================
--- libavformat/allformats.c	(revision 13820)
+++ libavformat/allformats.c	(working copy)
@@ -100,7 +100,7 @@
     REGISTER_MUXDEMUX (MATROSKA, matroska);
     REGISTER_MUXER    (MATROSKA_AUDIO, matroska_audio);
     REGISTER_MUXDEMUX (MJPEG, mjpeg);
-    REGISTER_DEMUXER  (MLP, mlp);
+    REGISTER_MUXDEMUX (MLP, mlp);
     REGISTER_DEMUXER  (MM, mm);
     REGISTER_MUXDEMUX (MMF, mmf);
     REGISTER_MUXDEMUX (MOV, mov);
