Index: libavformat/Makefile
===================================================================
--- libavformat/Makefile	(revision 13535)
+++ libavformat/Makefile	(working copy)
@@ -80,6 +80,7 @@
 OBJS-$(CONFIG_MATROSKA_MUXER)            += matroskaenc.o matroska.o riff.o avc.o
 OBJS-$(CONFIG_MJPEG_DEMUXER)             += raw.o
 OBJS-$(CONFIG_MJPEG_MUXER)               += raw.o
+OBJS-$(CONFIG_MLP_DEMUXER)               += raw.o
 OBJS-$(CONFIG_MM_DEMUXER)                += mm.o
 OBJS-$(CONFIG_MMF_DEMUXER)               += mmf.o raw.o
 OBJS-$(CONFIG_MMF_MUXER)                 += mmf.o riff.o
Index: libavformat/raw.c
===================================================================
--- libavformat/raw.c	(revision 13535)
+++ libavformat/raw.c	(working copy)
@@ -442,6 +442,19 @@
     .value = CODEC_ID_SHORTEN,
 };
 
+AVInputFormat mlp_demuxer = {
+    "mlp",
+    "raw mlp",
+    0,
+    NULL,
+    audio_read_header,
+    raw_read_partial_packet,
+    raw_read_close,
+    .flags= AVFMT_GENERIC_INDEX,
+    .extensions = "mlp",
+    .value = CODEC_ID_MLP,
+};
+
 AVInputFormat flac_demuxer = {
     "flac",
     "raw flac",
Index: libavformat/allformats.c
===================================================================
--- libavformat/allformats.c	(revision 13535)
+++ libavformat/allformats.c	(working copy)
@@ -100,6 +100,7 @@
     REGISTER_MUXDEMUX (MATROSKA, matroska);
     REGISTER_MUXER    (MATROSKA_AUDIO, matroska_audio);
     REGISTER_MUXDEMUX (MJPEG, mjpeg);
+    REGISTER_DEMUXER  (MLP, mlp);
     REGISTER_DEMUXER  (MM, mm);
     REGISTER_MUXDEMUX (MMF, mmf);
     REGISTER_MUXDEMUX (MOV, mov);
